staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -97,6 +111,40 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 	return resp, ok, err
 }
 
-func keyFunc(auds []string, token string) string {
-	return fmt.Sprintf("%#v|%v", auds, token)
+// keyFunc generates a string hash based on the inputs.
+// This lowers the memory requirement of the cache and keeps tokens out of memory.
+func keyFunc(randomCachePrefix []byte, auds []string, token string) string {
+	// allocate a single buffer based on the longest string
+	maxLength := len(token)
+	for _, aud := range auds {
+		if length := len(aud); length > maxLength {
+			maxLength = length
+		}
+	}
+	b := make([]byte, 4+maxLength) // include extra bytes to store the length
+
+	h := sha256.New()
```

Comment:
You can use sync.Pool + h.Reset to get rid of this alloc.

https://golang.org/pkg/sync/#Pool

Reply:
(unclear if the map lookup + lock would be faster, though)